src/FeatureSet.php

```diff
@@ -121,6 +123,12 @@ public function getTimeProvider()
         return $this->timeProvider;
     }
 
+    public function setTimeProvider(TimeProviderInterface $timeProvider)
+    {
+        $this->timeProvider = $timeProvider;
+        $this->timeGenerator = $this->buildTimeGenerator();
```

Comment:
You're calling this as the last line of the constructor, too. Does it also need to be here?